core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamTest.java

```diff
@@ -0,0 +1,99 @@
+package com.baeldung.inputstream;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertLinesMatch;
+import static org.junit.jupiter.api.io.CleanupMode.ALWAYS;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class InputStreamTest {
+
+    private Path sharedTempDir = null;
+    @Test
+    public void givenAString_whenWrittenToFileInputStream_thenShouldMatchWhenRead(@TempDir(cleanup = ALWAYS) Path tempDir) throws IOException {
+        sharedTempDir = tempDir;
+        Path sampleOut = sharedTempDir.resolve("sample-out.txt");
+        List<String> lines = Arrays.asList("Hello. This is just a test. Good bye.");
+        Files.write(sampleOut, lines);
+        File inputStreamFile = sampleOut.toFile();
+        try(InputStream inputStream = new FileInputStream(inputStreamFile)){
+            Assert.assertTrue(readString(inputStream).contains(lines.get(0)));
+        }
+    }
+
+    @Test
+    public void givenAString_whenWrittenToByteArrayInputStream_thenShouldMatchWhenRead() throws IOException {
+        String sample = "In this example, we use btye array input stream";
+        byte[] byteArray = sample.getBytes(StandardCharsets.UTF_8);
+        try(InputStream inputStream = new ByteArrayInputStream(byteArray)){
+            Assert.assertTrue(readString(inputStream).contains(sample));
+        }
+
+    }
+
+
+    @Test
+    public void readWithObjectInputStream(@TempDir(cleanup = ALWAYS) Path tempDir) throws IOException, ClassNotFoundException {
+        sharedTempDir = tempDir;
+        Path sampleOut = sharedTempDir.resolve("sample-out.txt");
+        File fileText = sampleOut.toFile();
+        //Serialize a Hashmap then write it into a file.
+        Map<String, String> kv = new HashMap<>();
+        try(  FileOutputStream fileOutStream = new FileOutputStream( fileText)){
```

Comment:
Fix the formatting - please apply the Baeldung code formatter, which you can find in the style guidelines for your IDE